lib/IRGen/IRGenSIL.cpp

```diff
@@ -2266,6 +2266,13 @@ Callee LoweredValue::getCallee(IRGenFunction &IGF,
   switch (kind) {
   case Kind::FunctionPointer: {
     auto &fn = getFunctionPointer();
+    if (calleeInfo.OrigFnType->getRepresentation() ==
+        SILFunctionTypeRepresentation::ObjCMethod) {
+      // Direct calls to Objective-C methods have a selector value of `undef`.
+      auto selectorValue =
+          llvm::UndefValue::get(fn.getFunctionType()->getParamType(1));
+      return Callee(std::move(calleeInfo), fn, selfValue, selectorValue);
```

Comment:
Perhaps it's better to resolve this with a new `SILFunctionTypeRepresentation::ObjCDirectMethod`. @rjmccall, what do you think?

Reply:
Let me know if you'd like to have a `SILFunctionTypeRepresnetation::ObjCDirectMethod` and I can add it.

Reply:
Instead of adding a new SILFunctionType representation, I would suggest a new CalleeKind instead.

Reply:
Yeah, I agree with Slava.

Reply:
I've added `getObjCDirectMethodCallee()` to generate the callee here.